Python/compile.c

```diff
@@ -5782,7 +5782,7 @@ compiler_comprehension(struct compiler *c, expr_ty e, int type,
     co = optimize_and_assemble(c, 1);
     compiler_exit_scope(c);
     if (is_top_level_await && is_async_generator){
-        c->u->u_ste->ste_coroutine = 1;
+        assert(c->u->u_ste->ste_coroutine == 1);
```

Comment:
I believe this assignment has been obsolete for a while now, see https://github.com/python/cpython/pull/6766#issuecomment-2204510340.